lib/http1.c

```diff
@@ -177,18 +177,26 @@ static void set_req_timeout(struct st_h2o_http1_conn_t *conn, uint64_t timeout,
 {
     if (conn->_timeout_entry.cb != NULL)
         h2o_timer_unlink(&conn->_timeout_entry);
-    conn->_timeout_entry.cb = cb;
-    if (cb != NULL)
-        h2o_timer_link(conn->super.ctx->loop, timeout, &conn->_timeout_entry);
+    if (conn->req.is_tunnel_req) {
+        conn->_timeout_entry.cb = NULL;
+    } else {
+        conn->_timeout_entry.cb = cb;
+        if (cb != NULL)
+            h2o_timer_link(conn->super.ctx->loop, timeout, &conn->_timeout_entry);
+    }
```

Comment:
Maybe it makes sense to do something like
```c
    if (!conn->req.is_tunnel_req)
        cb = NULL;
```
at the very beginning of the function and keep the rest as-is, considering that this function is invoked with `cb` set to NULL in order to clear the timeout.

Reply:
Is
```c
if (conn->req.is_tunnel_req)
    cb = NULL;
```
the right one?

Reply:
Applied the suggestion in https://github.com/h2o/h2o/pull/3129/commits/cfbe0a50d464a5dac5bbb7ab7c2d01410c9648e7.